internal/pkg/idgen/idgen.go

```diff
@@ -1,21 +1,28 @@
 package idgen
 
-// package naming is wrong.
-
 import (
 	"github.com/sony/sonyflake"
 )
 
-var flake *sonyflake.Sonyflake
+type RandomIdGenerator interface {
```

Comment:
changes in this pkg, make sure it not break other package that use this.

Reply:
Sure. we have taken care of it.